src/rmt.rs

```diff
@@ -856,3 +915,189 @@ mod chip {
         }
     }
 }
+
+use self::config::ReceiveConfig;
+/// The RMT receiver.
+///
+/// Use [`RxRmtDriver::start()`] to receive pulses.
+///
+/// See the [rmt module][crate::rmt] for more information.
+use alloc::vec::Vec;
+use core::convert::TryInto;
+use core::ptr;
+
+pub struct RxRmtDriver<'d> {
+    channel: u8,
+    _p: PhantomData<&'d mut ()>,
+    pub pulse_pair_vec: Vec<PulsePair>,
+}
+
+impl<'d> RxRmtDriver<'d> {
+    /// Initialise the rmt module with the specified pin, channel and configuration.
+    ///
+    /// To uninstall the driver just drop it.
+    ///
+    /// Internally this calls `rmt_config()` and `rmt_driver_install()`.
+
+    pub fn new<C: RmtChannel>(
+        _channel: impl Peripheral<P = C> + 'd,
+        pin: impl Peripheral<P = impl InputPin> + 'd,
+        config: &ReceiveConfig,
+        rx_buffer_size_in_bytes: u32,
```

Comment:
This should be `usize`, not `u32`. I also suggest to change the semantics of this to be size-in-PulsePair-instances, i.e. rename to just `rx_buffer_size` and do the math to bytes inside the constructor.